src/main/java/org/mule/extension/ftp/internal/connection/FtpAbstractConnectionProvider.java

```diff
@@ -1,350 +0,0 @@
- * LICENSE.txt file.
- */
-package org.mule.extension.ftp.internal.connection;
-
-import static java.lang.String.format;
-import static org.mule.extension.file.common.api.exceptions.FileError.CANNOT_REACH;
-import static org.mule.extension.file.common.api.exceptions.FileError.CONNECTION_TIMEOUT;
-import static org.mule.extension.file.common.api.exceptions.FileError.CONNECTIVITY;
-import static org.mule.extension.file.common.api.exceptions.FileError.INVALID_CREDENTIALS;
-import static org.mule.extension.file.common.api.exceptions.FileError.SERVICE_NOT_AVAILABLE;
-import static org.mule.extension.file.common.api.exceptions.FileError.UNKNOWN_HOST;
-import static org.mule.runtime.extension.api.annotation.param.ParameterGroup.CONNECTION;
-import static org.mule.runtime.extension.api.annotation.param.display.Placement.ADVANCED_TAB;
-import static org.slf4j.LoggerFactory.getLogger;
-
-import org.mule.extension.file.common.api.FileSystemProvider;
-import org.mule.extension.file.common.api.exceptions.FileError;
-import org.mule.extension.ftp.api.FTPConnectionException;
-import org.mule.extension.ftp.api.ftp.FtpTransferMode;
-import org.mule.extension.ftp.internal.TimeoutSettings;
-import org.mule.extension.ftp.internal.logging.LoggingOutputStream;
-import org.mule.runtime.api.connection.ConnectionException;
-import org.mule.runtime.api.connection.ConnectionValidationResult;
-import org.mule.runtime.api.connection.PoolingConnectionProvider;
-import org.mule.runtime.api.lifecycle.Initialisable;
-import org.mule.runtime.api.lock.LockFactory;
-import org.mule.runtime.extension.api.annotation.param.Optional;
-import org.mule.runtime.extension.api.annotation.param.Parameter;
-import org.mule.runtime.extension.api.annotation.param.ParameterGroup;
-import org.mule.runtime.extension.api.annotation.param.display.DisplayName;
-import org.mule.runtime.extension.api.annotation.param.display.Placement;
-import org.mule.runtime.extension.api.annotation.param.display.Summary;
-import org.mule.runtime.extension.api.annotation.values.OfValues;
-import org.mule.sdk.api.annotation.semantics.connectivity.ExcludeFromConnectivitySchema;
-
-import java.io.OutputStreamWriter;
-import java.io.PrintWriter;
-import java.io.UnsupportedEncodingException;
-import java.net.ConnectException;
-import java.net.SocketTimeoutException;
-import java.net.UnknownHostException;
-import java.util.concurrent.TimeUnit;
-import java.util.concurrent.atomic.AtomicBoolean;
-import java.util.function.Consumer;
-
-import javax.inject.Inject;
-
-import org.apache.commons.net.PrintCommandListener;
-import org.apache.commons.net.ftp.FTPClient;
-import org.apache.commons.net.ftp.FTPReply;
-import org.slf4j.Logger;
-
-/**
- * Connects to an FTP server
- *
- * @since 1.6.0
- */
-@DisplayName("FTP Connection")
-@Summary("Connection to connect against an FTP server")
-public abstract class FtpAbstractConnectionProvider extends FileSystemProvider<FtpFileSystem> implements
```

Comment:
We need this file for FTPS Connector. We'll face issues if we extend FtpConnectionProvider directly in ftps-connector.

Reply:
Reverted FileSystemProvider, FtpAbstractConnectionProvider and FtpConnectionProvider hierarchy to same as previous.